embassy-rp/src/clocks.rs

```diff
@@ -2,10 +2,139 @@ use pac::clocks::vals::*;
 
 use crate::{pac, reset};
 
-const XOSC_MHZ: u32 = 12;
+// TODO fix terrible use of global here
+static mut XIN_HZ: u32 = 0;
+
+pub struct ClockConfig {
+    rosc_config: Option<RoscConfig>,
```

Comment:
i'd remove `_config` from the field names, if the struct is named `ClockConfig` it's obvious they're all config.

Reply:
Done